### ISSUE TITLE
API: Is there a way for a provider to determine if it has been chosen in Muzei

### ISSUE THREAD

**Issue by yellowbluesky** [romannurik/muzei#658]
I have been getting a number of user reports that they don't know how to setup and use my provider. I would like to notify my users that they have not chosen my provider in Muzei, and to provide to them a short tutorial on how to do so.

I don't know much about API design, but perhaps this could be implementing a isSelectedProvider() method in ProviderContract

**Comment by ianhanniballake** [comment]
`ProviderContract` is for talking to your own `MuzeiArtProvider`. You'd want to use the existing `MuzeiContract` APIs for getting information from Muzei.

Specifically, you'd want to look at the `MuzeiContract.Sources.CONTENT_URI` (available since Muzei 2.3) and query it for the `MuzeiContract.Sources.COLUMN_NAME_AUTHORITY` - that's the same authority that your `MuzeiArtProvider` uses:

```
fun isSelected(context: Context, authority: String)  = context.contentResolver.query(
        MuzeiContract.Sources.CONTENT_URI,
        arrayOf(MuzeiContract.Sources.COLUMN_NAME_AUTHORITY),
        null, null, null)?.use { data ->
    // Technically, this is always 0, but defensive programming
    val authorityColumn = data.getColumnIndex(MuzeiContract.Sources.COLUMN_NAME_AUTHORITY)
    // Technically, there is only one selected source at a time right now
    while (data.moveToNext()) {
        val selectedAuthority = data.getString(authorityColumn)
        if (selectedAuthority == authority) {
            return true
        }
    }
    false
} ?: false
```

We have a couple of helper methods already in `MuzeiContract` and this seems like a reasonable thing to add.

I think this would go really well with a feature I had been planning for quite some time (but finally added as an issue here) in #659 where you could deep link directly into Muzei's Sources screen, scrolling to your provider to have the user enable it.

**Comment by yellowbluesky** [comment]
Sounds good, and the deep linking idea is great.

Thanks for your fantastic work 👍
